lib/file_store/base_store.rb

```diff
@@ -9,7 +9,7 @@ def store_upload(file, upload, content_type = nil)
       store_file(file, path)
     end
 
-    def store_optimized_image(file, optimized_image)
+    def store_optimized_image(file, optimized_image, content_type = nil, secure: false)
```

Comment:
Minor thing: do we have to define `content_type` and `secure` here? They aren't used when storing locally.

Reply:
The problem is @pmusaraj both the methods had a different signature, but the S3 has the additional secure + content_type parameters. In `OptimizedImage` I had:

```
url = Discourse.store.store_optimized_image(file, thumbnail, nil, secure: upload.secure?)
```

But this would fail all the local specs. I could add a conditional here e.g. if its S3 store use the extra params, but I thought it would be better for the method signature to be the same if we are overriding it. I guess the other alternative is just adding `opts` to the base_store method and using those where needed.

Reply:
That makes sense, thanks!